### PR TITLE
ensure TestSelfConnect doesn't hang forever

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Fixes `TestSelfConnect` to have a timeout

### Motivation

It can hang forever if the child process doesn't output a newline.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by hmahmood** [comment]
Why wouldn't `fork.py` produce a newline?

**Comment by brycekahle** [comment]
> Why wouldn't `fork.py` produce a newline?

I'm not sure what is causing this situation to manifest, but it has happened twice in CI today.

**Comment by brycekahle** [comment]
To clarify, I'm not sure that a newline isn't being produced, but this line is not returning `portStr, err := bufio.NewReader(stdOutReader).ReadString('\n')`

File: pkg/network/tracer/tracer_linux_test.go [hunk 1]
```diff
@@ -1013,29 +1014,26 @@ func (s *TracerSuite) TestSelfConnect() {
...
+
+	cmd := exec.CommandContext(ctx, "testdata/fork.py")
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+	cmd.WaitDelay = 10 * time.Second
+	stdOutReader, err := cmd.StdoutPipe()
+	require.NoError(t, err)
+	require.NoError(t, cmd.Start())
+	t.Cleanup(func() {
+		cancel()
```
Comment (hmahmood) [inline pkg/network/tracer/tracer_linux_test.go:1028]: Isn't this already called by line 1018?

Comment (brycekahle) [reply pkg/network/tracer/tracer_linux_test.go:1028]: Yes, but we need it to be called before the `Wait`, not after.

**Review by hmahmood — APPROVED** [review]

**Pull request merged**